src/emit/strings.rs

```diff
@@ -183,27 +188,42 @@ pub(super) fn format_string<'a, T: TargetRuntime<'a> + ?Sized>(
                     };
                 }
                 Type::Address(_) | Type::Contract(_) => {
-                    // for Solana/Substrate, we should encode in base58
+                    // For substrate we should encode in the SS58 format
```

Comment:
Should be a FIXME

Reply:
On Solana, it would be good if `"address_in_hex:{:x}".format(addr)` would still work

Reply:
Fixed